dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/event/listener/ServiceInstancesChangedListener.java

```diff
@@ -84,8 +82,8 @@
     public ServiceInstancesChangedListener(Set<String> serviceNames, ServiceDiscovery serviceDiscovery) {
         this.serviceNames = serviceNames;
         this.serviceDiscovery = serviceDiscovery;
-        this.listeners = new ConcurrentHashMap<>();
-        this.listenerQueue = new ConcurrentLinkedQueue<>();
+        this.listeners = new HashMap<>();
+        this.listenerQueue = new LinkedList<>();
```

Comment:
`listenerQueue` will be used concurrency in `notifyAddressChanged` method.

Reply:
Currently the `notifyAddressChanged` method is only called by the `doOnEvent` method with the `synchronized` modifier. The concurrent scenario you are talking about is that `notifyAddressChanged` may be called by subclasses in the future?